File: Jan/Week 3/LeetCode/problem_4.py
# Day 18 Problem No 39

# 643. Maximum Average Subarray I => https://leetcode.com/problems/maximum-average-subarray-i/description/?envType=study-plan-v2&envId=leetcode-75

class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        currS = sum(nums[:k])
        maxS = currS

        for i in range(k,len(nums)):
            currS += nums[i] - nums[i-k]
            maxS = max(maxS,currS)
        return maxS/k
         
        

File: Jan/Week 4/LeetCode/Problem_4.cpp
// Day 24 Problem No 52

// 1679. Max Number of K-Sum Pairs => https://leetcode.com/problems/max-number-of-k-sum-pairs/description/?envType=study-plan-v2&envId=leetcode-75

class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        sort(nums.begin(),nums.end());
        int l = 0;
        int r = nums.size() - 1;
        int opt = 0;
        while(l < r){
            if(nums[l]+nums[r] == k){
                opt++;
                l++;
                r--;
            }
            else if(nums[l]+nums[r] < k){
                l++;
            }
            else{
                r--;
            }
        }
        return opt;
    }
};